### ISSUE TITLE
Error in `stake permit` test: `AssertionError: expected false to equal true` 

### ISSUE THREAD

**Issue by matheusaaguiar** [euler-xyz/euler-contracts#119]
Recently,  this test started failing:
```
 338 passing (7m)
  1 failing

  1) staking
       stake permit:

      AssertionError: expected false to equal true
      + expected - actual

      -false
      +true
      
      at Object.cb (test/eulStakes.js:522:55)
      at TestSet._runAction (test/lib/eTestLib.js:1191:13)
      at TestSet._runTest (test/lib/eTestLib.js:1044:26)
      at Context.<anonymous> (test/lib/eTestLib.js:1016:21)
```
[1068599](https://app.circleci.com/pipelines/github/ethereum/solidity/24304/workflows/9f72146d-e6dd-4dac-84e6-02deb1c95791/jobs/1068599) and [1068676](https://app.circleci.com/pipelines/github/ethereum/solidity/24306/workflows/c2603542-8859-4425-9b59-07bd877e8e4f/jobs/1068676) are two examples of the problem in CI,  but it is also reproduced in a local run:
 
```
git clone https://github.com/euler-xyz/euler-contracts
cd euler-contracts/
npm init
npm --save-dev install hardhat
npx hardhat test test/eulStakes.js
```

The problem seems to be related to 4 occurrences of `et.expect(result.domainSeparatorMatch).to.equal(true);` in file `eulStake.js`. 
When `true` is changed to `false`, the test is successful.

**Comment by hoytech** [comment]
Thanks for the report! We were just talking about this. @kasperpawlowski determined that an upgrade of `@ethereumjs/vm` from `5.7.0` to `5.9.0` seems to have changed the behaviour of the chainId op-code. We're not sure why yet.

A short-term workaround is to change the `networks.hardhat.chainId` setting in `hardhat.config.js` from `1` to `0`.

**Comment by matheusaaguiar** [comment]
> A short-term workaround is to change the `networks.hardhat.chainId` setting in `hardhat.config.js` from `1` to `0`.

Thanks for that suggestion, but, unfortunately, this change solves the problem in the `stake permit` test, but then looks like it disrupts 3 other tests (local run output): 
```
  3 failing

  1) permit
       EIP2612 standard:
     Error: VM Exception while processing transaction: reverted with reason string 'permit: unauthorized'
    at Proxy.<fallback> (contracts/Proxy.sol:42)
    at HardhatNode._mineBlockWithPendingTxs (node_modules/hardhat/src/internal/hardhat-network/provider/node.ts:1773:23)
    at HardhatNode.mineBlock (node_modules/hardhat/src/internal/hardhat-network/provider/node.ts:466:16)
    at EthModule._sendTransactionAndReturnHash (node_modules/hardhat/src/internal/hardhat-network/provider/modules/eth.ts:1504:18)
    at HardhatNetworkProvider.request (node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:118:18)
    at EthersProviderWrapper.send (node_modules/@nomiclabs/hardhat-ethers/src/internal/ethers-provider-wrapper.ts:13:20)
  

  2) permit
       packed type:
     Error: VM Exception while processing transaction: reverted with reason string 'permit: unauthorized'
    at Proxy.<fallback> (contracts/Proxy.sol:42)
    at HardhatNode._mineBlockWithPendingTxs (node_modules/hardhat/src/internal/hardhat-network/provider/node.ts:1773:23)
    at HardhatNode.mineBlock (node_modules/hardhat/src/internal/hardhat-network/provider/node.ts:466:16)
    at EthModule._sendTransactionAndReturnHash (node_modules/hardhat/src/internal/hardhat-network/provider/modules/eth.ts:1504:18)
    at HardhatNetworkProvider.request (node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:118:18)
    at EthersProviderWrapper.send (node_modules/@nomiclabs/hardhat-ethers/src/internal/ethers-provider-wrapper.ts:13:20)
  

  3) permit
       allowed type:
     Error: VM Exception while processing transaction: reverted with reason string 'permit: unauthorized'
    at Proxy.<fallback> (contracts/Proxy.sol:42)
    at HardhatNode._mineBlockWithPendingTxs (node_modules/hardhat/src/internal/hardhat-network/provider/node.ts:1773:23)
    at HardhatNode.mineBlock (node_modules/hardhat/src/internal/hardhat-network/provider/node.ts:466:16)
    at EthModule._sendTransactionAndReturnHash (node_modules/hardhat/src/internal/hardhat-network/provider/modules/eth.ts:1504:18)
    at HardhatNetworkProvider.request (node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:118:18)
    at EthersProviderWrapper.send (node_modules/@nomiclabs/hardhat-ethers/src/internal/ethers-provider-wrapper.ts:13:20)
```

I guess for now I will change the `eulStake.js` file directly since that fixes the problem without interfering with other tests.

**Comment by kasperpawlowski** [comment]
Hey guys! Sorry for the delay. I investigated the root of the issue and I'm pretty confident that the problem originates from the `hardhat-core` package. I opened an issue ticket a few days ago, but I haven't heard from the hardhat team yet. Anyways, I'm sharing that, we'll see where it goes.

https://github.com/NomicFoundation/hardhat/issues/2753

**Comment by kasperpawlowski** [comment]
Hey guys! My assumption was correct, the failing test originated from `hardhat-core` package bug. The bug was fixed in [v2.9.7](https://github.com/NomicFoundation/hardhat/releases/tag/hardhat%402.9.7). Please update hardhat and check if everything works for you. If it does, it seems that we can close the issue.

**Comment by matheusaaguiar** [comment]
Sorry for the late response. It works now. Issue solved.
